bin/fsl_sub

```diff
@@ -488,58 +527,99 @@ EOF2
 	echo $$
 	;;
 
+
+#############################
+# CBRAIN ADDENDUM #2
+# Support for method "CBRAIN"
+#############################
+
     CBRAIN)
+
+        # This function preps a set of files that
+        # CBRAIN will recuperate, and cause the creation
+        # of a new CBRAIN task of type 'FslSub'.
+        #
+        # The function receives in $1 a number, usually an
+        # arbitrary task counter, and in the rest of the arguments
+        # a full command to run. E.g.
+        #
+        #    submit_cbrain 2 ls -l
+        #
+        # The function will create a ".sh" file with the command, and
+        # a ".json" descriptor file for CBRAIN to pick up and process.
         function submit_cbrain {
+
+          cbrain_subtask_number="$1"; shift;
+
           if [ $verbose -eq 1 ] ; then
-            echo Submitting to CBRAIN: $* >&2
+            echo Submitting subtask $cbrain_subtask_number to CBRAIN: $* >&2
           fi
-          # Uses CBRAIN's mechanism for 
-          # CBRAIN tasks to submit new tasks
-          FILE=`mktemp ${CBRAIN_WORKDIR}/new-task-XXXX.json`
-          cat << NEWTASK > ${FILE}
+
+          pushd "$CBRAIN_WORKDIR" >/dev/null # save current workdir and switch to CBRAIN's
+
+          # Prepare an ID to uses CBRAIN's mechanism for
+          # submit new subtasks.
+          sub_id=`mktemp ${cbrain_subtask_number}_XXXX`  # e.g. "0_u3JK"
+          rm -f "$sub_id" # handle non-needed empty file just created
+          prefix=".new-task-$sub_id" # used for both .json and .sh files
+
+          # Create the bash script
+          cat <<BASH_SCRIPT >"$prefix.sh"
+#
+# This file created automatically by $0
+#
+# This is the code for CBRAIN local subtask $prefix
+#
+
+$@
+
+BASH_SCRIPT
+
+          # Create the JSON descriptor
+          cat <<JSON_FILE >"$prefix.json"
 {
 "tool-class": "CbrainTask::FslSub",
-"description": "$*",
+"description": "$JobName",
+"share_wd_tid": "$CBRAIN_SHARE_WD_TID",
```

Comment:
Would be cleaner if it was called 'share-wd-tid' (but has to be modified consistently in cluster_task.rb, see comment in https://github.com/aces/cbrain/pull/355).